chainstate/src/interface/chainstate_interface.rs

```diff
@@ -127,11 +129,31 @@ pub trait ChainstateInterface: Send + Sync {
         headers: Vec<SignedBlockHeader>,
     ) -> Result<(Vec<SignedBlockHeader>, Vec<SignedBlockHeader>), ChainstateError>;
 
-    fn get_block_index(&self, id: &Id<Block>) -> Result<Option<BlockIndex>, ChainstateError>;
-    fn get_gen_block_index(
+    /// Return the block index given a block id.
+    /// This function will only return persistent block indices; if a non-persistent index
```

Comment:
I think a good definition of "persistent" or "persisted" (in case they were renamed) is needed. Can we add a definition here in the lines of "a persisted block index is a block index whose block is stored in the database"

Reply:
> I think a good definition of "persistent" or "persisted" (in case they were renamed) is needed. Can we add a definition here in the lines of "a persisted block index is a block index whose block is stored in the database"

As we've discussed, now I use the word "persisted" but it refers to the block itself rather than the index object. So, the fact that the block data has to exist is kind of self-explanatory now:
```
/// Return the block index given a block id.
/// This function will only return block indices for persisted blocks; if the corresponding
/// block hasn't been persisted, None will be returned.
fn get_block_index_for_persisted_block(
```
Instead, I've added a sentence to the comments for the "any" variants, mentioning that the index object may not live long:
```
/// Return the block index given a block id. The corresponding block may or may not be persisted.
/// Note that a block index object for a non-persisted block may be deleted from the db at any moment.
fn get_gen_block_index_for_any_block(
```